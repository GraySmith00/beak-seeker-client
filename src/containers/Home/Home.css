.home {
  display: flex;
  flex-direction: column;
  align-items: center; }
  .home .main-content {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-top: 80px;
    overflow: scroll;
    width: 100%; }
  .home .nav-link {
    display: flex;
    justify-content: center;
    margin: 1rem auto;
    padding: 1rem;
    width: 80%;
    background: crimson;
    text-decoration: none;
    color: #fff;
    border-radius: 5px;
    -webkit-box-shadow: 10px 10px 26px -9px rgba(0, 0, 0, 0.43);
    -moz-box-shadow: 10px 10px 26px -9px rgba(0, 0, 0, 0.43);
    box-shadow: 10px 10px 26px -9px rgba(0, 0, 0, 0.43); }
