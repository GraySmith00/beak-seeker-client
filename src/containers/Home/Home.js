import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';

import { setCurrentUser } from '../../actions/thunks/setCurrentUser';

export class Home extends Component {
  componentDidMount() {
    const id = window.location.search.slice(4);

    this.props.setCurrentUser(id);
  }

  render() {
    return (
      <div className="home">
        <nav>
          <NavLink to="/hotspots">Nearby Hotspots</NavLink>
          <NavLink to="/leaderboard">Leaderboard</NavLink>
        </nav>
      </div>
    );
  }
}

Home.propTypes = {
  setCurrentUser: PropTypes.func.isRequired
};

const mapDispatchToProps = dispatch => ({
  setCurrentUser: id => dispatch(setCurrentUser(id))
});

export default connect(
  null,
  mapDispatchToProps
)(Home);
