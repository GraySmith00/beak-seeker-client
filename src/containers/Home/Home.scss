.home {
  display: flex;
  flex-direction: column;
  align-items: center;

  margin-bottom: 2rem;
  background: url('./birdBackground2.jpg');
  background-size: cover;
  background-position: top center;
  height: 100%;

  .main-content {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    margin-top: 90px;

    overflow: scroll;
    width: 100%;
  }

  .greeting {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  .greeting img {
    width: 100px;
    border-radius: 50%;
  }

  .nav-link {
    display: flex;
    align-items: center;
    justify-content: center;

    margin: 1rem auto;
    padding: 1rem;

    width: 80%;
    background: rgba(0, 0, 0, 0.75);
    text-decoration: none;
    color: #fff;
    border-radius: 5px;
    font-size: 1.3rem;

    -webkit-box-shadow: 10px 10px 26px -9px rgba(0, 0, 0, 0.43);
    -moz-box-shadow: 10px 10px 26px -9px rgba(0, 0, 0, 0.43);
    box-shadow: 10px 10px 26px -9px rgba(0, 0, 0, 0.43);
  }

  i {
    margin-right: 10px;
  }
}
