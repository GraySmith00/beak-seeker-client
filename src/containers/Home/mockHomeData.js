export const mockUser = {
  sightings: [],
  _id: '5b98223ce17b1dbfc1b26ada',
  email: 'greg.smith00@gmail.com',
  username: 'greg_savoy',
  image:
    'https://pbs.twimg.com/profile_images/1001553154146615296/Jab4V7G0_normal.jpg',
  __v: 0
};
