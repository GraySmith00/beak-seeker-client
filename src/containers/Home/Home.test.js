import React from 'react';
import { shallow } from 'enzyme';
import { Home, mapDispatchToProps } from './Home';

import { mockUser } from './mockHomeData';

jest.mock('../../actions/thunks/setCurrentUser');

describe('Home component', () => {
  let wrapper;
  let setCurrentUser;

  beforeEach(() => {
    setCurrentUser = jest.fn();
    wrapper = shallow(<Home setCurrentUser={setCurrentUser} />);
  });

  it('shuould match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    it('should dispatch setCurrentUser using a function from mapDispatchToProps', async () => {
      // setup
      const mockDispatch = jest.fn();
      const acitonToDispatch = setCurrentUser(mockUser._id);

      // expectution
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setCurrentUser(mockUser._id);

      // expectation
      expect(mockDispatch).toHaveBeenCalledWith(acitonToDispatch);
    });
  });
});
