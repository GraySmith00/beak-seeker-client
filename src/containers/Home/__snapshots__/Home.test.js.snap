// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component shuould match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    setCurrentUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter()
          currentPage={
                    <p>
                              <i
                                        className="fas fa-home"
                              />
                               Home
                    </p>
          }
/>,
        <main
          className="main-content"
>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="nav-link"
                    to="/hotspots"
          >
                    <i
                              className="fas fa-map-marked-alt"
                    />
                     Nearby Hotspots
          </NavLink>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="nav-link"
                    to="/leaderboard"
          >
                    <i
                              className="fas fa-trophy"
                    />
                     Leaderboard
          </NavLink>
</main>,
      ],
      "className": "home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentPage": <p>
            <i
                        className="fas fa-home"
            />
             Home
</p>,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav-link"
              to="/hotspots"
>
              <i
                            className="fas fa-map-marked-alt"
              />
               Nearby Hotspots
</NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav-link"
              to="/leaderboard"
>
              <i
                            className="fas fa-trophy"
              />
               Leaderboard
</NavLink>,
          ],
          "className": "main-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": Array [
                <i
                  className="fas fa-map-marked-alt"
/>,
                " Nearby Hotspots",
              ],
              "className": "nav-link",
              "to": "/hotspots",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-map-marked-alt",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Nearby Hotspots",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": Array [
                <i
                  className="fas fa-trophy"
/>,
                " Leaderboard",
              ],
              "className": "nav-link",
              "to": "/leaderboard",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-trophy",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Leaderboard",
            ],
            "type": [Function],
          },
        ],
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter()
            currentPage={
                        <p>
                                    <i
                                                className="fas fa-home"
                                    />
                                     Home
                        </p>
            }
/>,
          <main
            className="main-content"
>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="nav-link"
                        to="/hotspots"
            >
                        <i
                                    className="fas fa-map-marked-alt"
                        />
                         Nearby Hotspots
            </NavLink>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="nav-link"
                        to="/leaderboard"
            >
                        <i
                                    className="fas fa-trophy"
                        />
                         Leaderboard
            </NavLink>
</main>,
        ],
        "className": "home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentPage": <p>
              <i
                            className="fas fa-home"
              />
               Home
</p>,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav-link"
                to="/hotspots"
>
                <i
                                className="fas fa-map-marked-alt"
                />
                 Nearby Hotspots
</NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav-link"
                to="/leaderboard"
>
                <i
                                className="fas fa-trophy"
                />
                 Leaderboard
</NavLink>,
            ],
            "className": "main-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": Array [
                  <i
                    className="fas fa-map-marked-alt"
/>,
                  " Nearby Hotspots",
                ],
                "className": "nav-link",
                "to": "/hotspots",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-map-marked-alt",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " Nearby Hotspots",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": Array [
                  <i
                    className="fas fa-trophy"
/>,
                  " Leaderboard",
                ],
                "className": "nav-link",
                "to": "/leaderboard",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-trophy",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " Leaderboard",
              ],
              "type": [Function],
            },
          ],
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
